### PR TITLE
import mappings for zalando problem

### LINKED ISSUE #9911: Generate importMappings with zalando problem class when using API first development
##### **Overview of the feature request**

When users try to use Zalando Problem library in theirs `api.yml`:
```
responses:
  default:
    description: error occurred - see status code and problem object for more information.
    content:
      "application/problem+json":
        schema:
          $ref: 'https://opensource.zalando.com/problem/schema.yaml#/Problem'
```
And run `mvn generate-sources`, maven plugin `openapi-generator-maven-plugin` generates unneseccary java class `Problem` eventhough `org.zalando.Problem` is already presented in classpath in `org.zalando.problem:problem` jar.

##### **Motivation for or Use Case**

Normally, it is expected from a user perspective to reuse existing classes in classpath. 
Consider adding `<importMappings>Problem=org.zalando.problem.Problem</importMappings>` inside `<configuration>` tag in `pom.xml`, please. It would fix the thing.

Thank you in advance!

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by atomfrede**
* Updating open api plugin to 4.0.1
* Add new required dependency for `json-nullable` (because of updated templates in swagger templates)
* Add import mapping for zalando problem, such these classes are not generated

closes #9911 
-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by atomfrede** [comment]
The failing tests seems totally unrelated, any ideas why it is failing?

**Comment by DanielFran** [comment]
This is happenning since few days and it is intermittent, do not know why?

**Comment by ruddell** [comment]
I restarted the failing job and it passed.  There are some errors related to [logstash](https://travis-ci.org/jhipster/generator-jhipster/jobs/546481753#L3309) and [kafka](https://travis-ci.org/jhipster/generator-jhipster/jobs/546481753#L3358) but they didn't seem to cause an issue the second time the test ran

**Comment by atomfrede** [comment]
Thanks @ruddell. It is totally strange, I restarted the failed build two times already and they still failed.

File: generators/server/templates/build.gradle.ejs [hunk 1]
```diff
@@ -42,6 +39,9 @@ plugins {
     id "jacoco"
     id "com.google.cloud.tools.jib" version "<%= JIB_VERSION %>"
     id "com.gorylenko.gradle-git-properties" version "2.0.0"
+    <%_ if (enableSwaggerCodegen) { _%>
+    id "org.openapi.generator" version "4.0.1"
```
Comment (DanielFran) [inline generators/server/templates/build.gradle.ejs]: Can you update to 4.0.2

File: generators/server/templates/pom.xml.ejs [hunk 2]
```diff
@@ -133,7 +133,7 @@
         <jib-maven-plugin.version><%= JIB_VERSION %></jib-maven-plugin.version>
         <lifecycle-mapping.version>1.0.0</lifecycle-mapping.version>
 <%_ if (enableSwaggerCodegen) { _%>
-        <openapi-generator-maven-plugin.version>3.3.4</openapi-generator-maven-plugin.version>
+        <openapi-generator-maven-plugin.version>4.0.1</openapi-generator-maven-plugin.version>
```
Comment (DanielFran) [inline generators/server/templates/pom.xml.ejs]: Can you update to 4.0.2

File: generators/server/templates/pom.xml.ejs [hunk 3]
```diff
@@ -294,6 +294,13 @@
             <groupId>commons-io</groupId>
             <artifactId>commons-io</artifactId>
         </dependency>
+<%_ if (enableSwaggerCodegen) { _%>
+        <dependency>
+            <groupId>org.openapitools</groupId>
+            <artifactId>jackson-databind-nullable</artifactId>
+            <version>0.1.0</version>
```
Comment (DanielFran) [inline generators/server/templates/pom.xml.ejs]: Can you create a property version?

**Pull request merged**